nix_update/update.py

```diff
@@ -36,10 +36,15 @@ def replace_version(package: Package) -> bool:
     if changed:
         info(f"Update {old_version} -> {new_version} in {package.filename}")
         with fileinput.FileInput(package.filename, inplace=True) as f:
-            for line in f:
+            for i, line in enumerate(f, 1):
                 if package.new_version.rev:
                     line = line.replace(package.rev, package.new_version.rev)
-                print(line.replace(f'"{old_version}"', f'"{new_version}"'), end="")
+                if (
+                    package.version_position is None
```

Comment:
Mhm. It's nice that we are more precise but is this not going to be a problem for the a lot of case i.e. if a user does this:

```
let
  version = "1.0.0";
in stdenv.mkDerivation {
  name = "package-${version}";
}
```

I see these cases actually more often than multiple packages in a file.

Reply:
Maybe we should do the following. First an iteration where we check if the version line has our version and if this doesn't work it will backtrack and also replace other assignments.

Reply:
Good point. I've added the precheck iteration but this handling isn't as simple